examples/geomstats_in_pymanopt.py

```diff
@@ -0,0 +1,113 @@
+"""This very simple example demonstrates how geometries from geomstats can be
+used in pymanopt to perform optimization on manifolds. It uses the Riemannian
+steepest descent solver.
+
+The example currently requires installing the pymanopt HEAD from git:
+
+    pip install git+ssh://git@github.com/pymanopt/pymanopt.git
+"""
+
+import functools
+import os
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"  # NOQA
+
+import numpy as np
+from numpy import linalg as la, random as rnd
+from pymanopt import Problem
+from pymanopt.manifolds.manifold import Manifold
+from pymanopt.solvers import SteepestDescent
+
+from geomstats.geometry.hypersphere import Hypersphere
+
+
+def squeeze_ndarray(f):
+    """Decorator to remove singleton dimensions from a numpy.ndarray returned
+    by the decorated function.
+    """
+    @functools.wraps(f)
+    def inner(*args, **kwargs):
+        return np.squeeze(f(*args, **kwargs))
+    return inner
+
+
+class GeomstatsSphere(Manifold):
```

Comment:
Could this wrapper be extended to a Wrapper for any GeomstatsManifold? Quickly glancing at the interface in pymanopt, most of the interface is available, except for `egrad2rgrad`, `ehess2rhess`, `retr` and some information about the structure of the manifold e.g. `typicaldist`.

Reply:
This was definitely the idea for future examples as it's fairly straightforward to wrap the API. The question is where such an adapter class should live as I'd think it odd if a pymanopt-specific class popped up somewhere in the geomstats codebase since pymanopt is not (yet^^) a hard dependency.